java-checks/src/main/java/org/sonar/java/checks/DeadStoreCheck.java

```diff
@@ -0,0 +1,210 @@
+  priority = Priority.MAJOR)
+@ActivatedByDefault
+@SqaleSubCharacteristic(RulesDefinition.SubCharacteristics.DATA_RELIABILITY)
+@SqaleConstantRemediation("15min")
+public class DeadStoreCheck extends SubscriptionBaseVisitor {
+
+  @Override
+  public List<Tree.Kind> nodesToVisit() {
+    return ImmutableList.of(Tree.Kind.METHOD);
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    MethodTree methodTree = (MethodTree) tree;
+    if (methodTree.block() == null) {
+      return;
+    }
+
+    // Exclude try statements with finally as CFG is incorrect for those and lead to false positive
+    if (hasTryFinallyWithLocalVar(methodTree.block(), methodTree.symbol())) {
+      return;
+    }
+
+    Symbol.MethodSymbol methodSymbol = methodTree.symbol();
+    CFG cfg = CFG.build(methodTree);
+    LiveVariables liveVariables = LiveVariables.analyze(cfg);
+    for (CFG.Block block : cfg.blocks()) {
+      checkElements(Lists.reverse(block.elements()), liveVariables.getOut(block), methodSymbol);
+    }
+  }
+
+  private void checkElements(List<Tree> elements, Set<Symbol> blockOut, Symbol.MethodSymbol methodSymbol) {
+    Set<Symbol> out = new HashSet<>(blockOut);
+    Set<Tree> assignmentLHS = new HashSet<>();
+    for (Tree element : elements) {
+      Symbol symbol;
+      switch (element.kind()) {
+        case ASSIGNMENT:
+          ExpressionTree lhs = ((AssignmentExpressionTree) element).variable();
+          if (lhs.is(Tree.Kind.IDENTIFIER)) {
+            symbol = ((IdentifierTree) lhs).symbol();
+            if (isLocalVariable(symbol) && !out.contains(symbol)) {
+              createIssue(element, symbol);
+            }
+            assignmentLHS.add(lhs);
+            out.remove(symbol);
+          }
+          break;
+        case IDENTIFIER:
+          symbol = ((IdentifierTree) element).symbol();
+          if (!assignmentLHS.contains(element) && isLocalVariable(symbol)) {
+            out.add(symbol);
+          }
+          break;
+        case VARIABLE:
+          out = handleVariable(out, element);
+          break;
+        case NEW_CLASS:
+          ClassTree body = ((NewClassTree) element).classBody();
+          if (body != null) {
+            List<Symbol> usedLocalVar = getUsedLocalVarInSubTree(body, methodSymbol);
+            out.addAll(usedLocalVar);
+          }
+          break;
+        case LAMBDA_EXPRESSION:
+          LambdaExpressionTree lambda = (LambdaExpressionTree) element;
+          out.addAll(getUsedLocalVarInSubTree(lambda.body(), methodSymbol));
+          break;
+        case TRY_STATEMENT:
+          TryStatementTree tryStatement = (TryStatementTree) element;
+          for (CatchTree catchTree : tryStatement.catches()) {
+            out.addAll(getUsedLocalVarInSubTree(catchTree, methodSymbol));
+          }
+          break;
+        case POSTFIX_INCREMENT:
+        case POSTFIX_DECREMENT:
+        case PREFIX_INCREMENT:
+        case PREFIX_DECREMENT:
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this switch case number of lines from 9 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1151)